src/dialogs/config-flow/dialog-data-entry-flow.ts

```diff
@@ -339,20 +344,28 @@ class DataEntryFlowDialog extends LitElement {
   }
 
   private async _subscribeDataEntryFlowProgressed() {
-    if (this._unsubDataEntryFlowProgressed) {
+    if (this._unsubDataEntryFlowProgress) {
       return;
     }
-    this._unsubDataEntryFlowProgressed = subscribeDataEntryFlowProgressed(
-      this.hass.connection,
-      async (ev) => {
+    this._progress = undefined;
+    const unsubs = [
+      subscribeDataEntryFlowProgressed(this.hass.connection, (ev) => {
         if (ev.data.flow_id !== this._step?.flow_id) {
           return;
         }
         this._processStep(
           this._params!.flowConfig.fetchFlow(this.hass, this._step.flow_id)
         );
-      }
-    );
+        this._progress = undefined;
+      }),
+      subscribeDataEntryFlowProgress(this.hass.connection, (ev) => {
```

Comment:
Was it too hard in core to combine these subscriptions?

Reply:
I haven't even tried because it uses a generic "subscribe to core event" API. I can improve this later when adding custom step support. Want to have a clearer design first though.
There will be a follow up PR for the new designs